site/content/articles/external-container-registry-integration.adoc

```diff
@@ -0,0 +1,479 @@
+
+
+== Running images stored in an external registry
+
+Images residing in external registries can be run within OpenShift. This section describes the steps necessary to configure an OpenShift project to utilize images stored in external registries as well as the various methods for managing external images.
+
+=== Deploying an image using the oc new-app command
+
+The simplest method for deploying an image from an external registry is to use the *oc new-app* CLI command. This command will resolve the image and set up the necessary API objects based on metadata contained in the image.
+
+Execute the following command to *sample-app* built in an earlier section
+
+[source]
+----
+oc new-app --docker-image=external-registry.example.com:5000/publicrepo/sample-app:latest
+----
+
+The CLI will configure services based on exposed ports on the image and set up a DeploymentConfig to deploy the image to the project.
+
+=== Deploying an image using an ImageStream
+
+An link:https://docs.openshift.com/container-platform/4.2/openshift_images/images-understand.html#images-imagestream-use_images-understand[ImageStream] represents a virtual repository of related docker images and can be used to abstract the location of image from applications. Images stored in an external repository can be referenced using an ImageStream to relieve the burden from the underlying application.
+
+To create a new ImageStream for the existing image, execute the following command:
+
+[source]
+----
+oc import-image sample-app --from=external-registry.example.com:5000/publicrepo/sample-app --confirm
+----
+
+A new ImageStream called *sample-app* will be created and can be used to deploy a new application. Execute the following command to deploy create the application from the imported ImageStream:
+
+[source]
+----
+oc new-app --image-stream=sample-app
+----
+
+
+=== Deploy an Image using the location in the DeploymentConfig
+
+Images from external registries can be deployed in an application by directly specifying its' location in the DeploymentConfig API object in the containers' *image* property as shown below.
+
+[source]
+----
+"containers": [
+                    {
+                        "name": "sample-app",
+                        "image": "external-registry.example.com:5000/publicrepo/sample-app:latest",  <1>
+                        "ports": [
+                            {
+                                "containerPort": 8080,
+                                "protocol": "TCP"
+                            }
+                        ],
+                        "resources": {},
+                        "terminationMessagePath": "/dev/termination-log",
+                        "imagePullPolicy": "Always"
+                    }
+                ],
+----
+<1> The image to deploy
+
+NOTE: It is recommended that an ImageStream be used instead of a direct reference
```

Comment:
Could we show what the config looks like when using an imagestream in a DeploymentConfig reference?